Jennyf/core logging

jennyf19 opened the pull request:
Fix pii logging in core.

jennyf19 committed:
Add int values to PromptBehavior enums

jennyf19 committed:
update core log messages for pii logs and non-pii logs
address build warnings (add `async` to end of a few test methods)

jennyf19 committed:
check logging throughout adal and msal
add license information

jennyf19 committed:
Merge branch 'dev' into jennyf/coreLogging

henrik-me commented on core/src/Cache/CacheFallbackOperations.cs:
```diff
@@ -74,7 +81,9 @@ internal class CacheFallbackOperations
             if (rtItem == null)
             {
-                CoreLoggerBase.Default.Info("rtItem is null. Skipping MSAL RT cache write");
+                string msg = "rtItem is null. Skipping MSAL RT cache write";
```
>rtItem is null.  [](start = 30, length = 16)

No refresh token available. (the current text is not something imo that a developer would understand unless they look into our code) #Resolved

jennyf19 replied:
yes...definitely. Good catch. wasn't paying much attention to what was actually in the message. thx

---
In reply to: [206365110](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#discussion_r206365110) [](ancestors = 206365110)

henrik-me commented on core/src/Cache/CacheFallbackOperations.cs:
```diff
@@ -131,7 +140,10 @@ internal class CacheFallbackOperations
             catch (Exception ex)
             {
-                CoreLoggerBase.Default.Warning("GetAllAdalUsersForMsal failed due to Exception - " + ex);
+                string msg = "GetAllAdalUsersForMsal failed due to Exception - ";
```
>GetAllAdalUsersForMsal failed due to Exception -  [](start = 30, length = 49)

in general we should not use messages which are tied to our code, rather the message should say something that makes sense for a developer (example provided above as well). For this one it could be: 
An error occurred while searching and matching adal users in the msal cache. #Resolved

jennyf19 replied:
done. #Pending

henrik-me commented on core/src/Platforms/android/LegacyCachePersistance.cs:
```diff
@@ -50,8 +50,10 @@ byte[] ILegacyCachePersistance.LoadCache()
             {
-                CoreLoggerBase.Default.Warning("Failed to load cache: " + ex.Message);
-                CoreLoggerBase.Default.ErrorPii(ex);
+                string msg = "Failed to load cache: ";
```
>Failed to load cache: [](start = 30, length = 21)

An error occurred while reading the adal cache? #Resolved

jennyf19 replied:
done. #Resolved

henrik-me commented on core/src/Platforms/android/LegacyCachePersistance.cs:
```diff
@@ -71,8 +73,10 @@ void ILegacyCachePersistance.WriteCache(byte[] serializedCache)
             {
-                CoreLoggerBase.Default.Warning("Failed to save cache: " + ex.Message);
-                CoreLoggerBase.Default.ErrorPii(ex);
+                const string msg = "Failed to save cache: ";
```
>Failed [](start = 36, length = 6)

... as above. consider adding adal or msal where relevant (seems like this is adal given the legacy naming #Pending

jennyf19 replied:
done. #Pending

henrik-me commented:
>         public new static void SetAuthorizationResult(AuthorizationResult authorizationResultInput)

aside: would have been great if this was not needed

---
Refers to: core/src/Platforms/Android/EmbeddedWebview/EmbeddedWebUI.cs:70 in fa2f142. [](commit_id = fa2f142624d2c2c15688c7ab678c7ae591dcd73f, deletion_comment = False)

henrik-me commented on core/src/Platforms/Android/EmbeddedWebview/EmbeddedWebUI.cs:
```diff
@@ -76,7 +76,9 @@ public new static void SetAuthorizationResult(AuthorizationResult authorizationR
             }
             else
```
>else [](start = 12, length = 4)

would it make sense to raise an exception instead allowing the developer to do something about this situation? #Pending

jennyf19 replied:
i removed this...but i'll do a quick test in the morning before merging. nothing seems to be using this code. Looks like it's a duplicate method and should have been deleted here, as it's used elsewhere.

---
In reply to: [206366224](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#discussion_r206366224) [](ancestors = 206366224)

henrik-me commented:
>             throw new NotImplementedException();

imo: strange that this one throws an exception... do you know why? #Pending

---
Refers to: core/tests/Test.Microsoft.Identity.Core.Unit/TestLogger.cs:75 in fa2f142. [](commit_id = fa2f142624d2c2c15688c7ab678c7ae591dcd73f, deletion_comment = False)

henrik-me commented:
thanks for getting this done.  #Resolved

henrik-me reviewed (changes requested):
:clock1:

jennyf19 commented:
>         public new static void SetAuthorizationResult(AuthorizationResult authorizationResultInput)

i don't think it is.

---
In reply to: [409058051](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#issuecomment-409058051) [](ancestors = 409058051)

---
Refers to: core/src/Platforms/Android/EmbeddedWebview/EmbeddedWebUI.cs:70 in fa2f142. [](commit_id = fa2f142624d2c2c15688c7ab678c7ae591dcd73f, deletion_comment = False)

jennyf19 commented:
>             throw new NotImplementedException();

No idea. I will remove it. I don't know why it throws here. maybe was an auto-fill

---
In reply to: [409058738](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#issuecomment-409058738) [](ancestors = 409058738)

---
Refers to: core/tests/Test.Microsoft.Identity.Core.Unit/TestLogger.cs:75 in fa2f142. [](commit_id = fa2f142624d2c2c15688c7ab678c7ae591dcd73f, deletion_comment = False)

jennyf19 commented:
:sunglasses: :coffee: 

---
In reply to: [409058977](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#issuecomment-409058977) [](ancestors = 409058977)

jennyf19 committed:
address PR comments
- more specific log messages

jennyf19 committed:
Merge branch 'jennyf/coreLogging' of https://github.com/AzureAD/azure-activedirectory-library-for-dotnet into jennyf/coreLogging

jennyf19 committed:
Merge branch 'dev' into jennyf/coreLogging

henrik-me commented on core/src/Cache/CacheFallbackOperations.cs:
```diff
@@ -228,7 +243,10 @@ internal class CacheFallbackOperations
             catch (Exception ex)
             {
-                CoreLoggerBase.Default.Warning("GetAllAdalEntriesForMsal failed due to Exception - " + ex);
+                string msg = "GetAllAdalEntriesForMsal failed due to Exception - ";
```
>GetAllAdalEntriesForMsal [](start = 30, length = 24)

should probably be more readable as well. E.g.: s: An error occurred while searching for valid MSAL cache entries in ADAL cache.

henrik-me commented:
>         public new static void SetAuthorizationResult(AuthorizationResult authorizationResultInput)

Does that mean we are not using it anywhere?

---
In reply to: [409078170](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#issuecomment-409078170) [](ancestors = 409078170,409058051)

---
Refers to: core/src/Platforms/Android/EmbeddedWebview/EmbeddedWebUI.cs:70 in fa2f142. [](commit_id = fa2f142624d2c2c15688c7ab678c7ae591dcd73f, deletion_comment = False)

henrik-me commented:
>             throw new NotImplementedException();

It seems like the build and tests completed thus likely not even testing anything.

---
In reply to: [409078304](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#issuecomment-409078304) [](ancestors = 409078304,409058738)

---
Refers to: core/tests/Test.Microsoft.Identity.Core.Unit/TestLogger.cs:75 in fa2f142. [](commit_id = fa2f142624d2c2c15688c7ab678c7ae591dcd73f, deletion_comment = False)

henrik-me commented:
only remaining concern is the texts used to inform the developers about issues. Approving as this PR really doesn't have to do with this, however it would be great if you could validate and make a more developer/user friendly log message. (#Supportability)

henrik-me reviewed (approved):
:shipit:

jennyf19 commented:
>         public new static void SetAuthorizationResult(AuthorizationResult authorizationResultInput)

it didn't have any references. 

---
In reply to: [409092453](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#issuecomment-409092453) [](ancestors = 409092453,409078170,409058051)

---
Refers to: core/src/Platforms/Android/EmbeddedWebview/EmbeddedWebUI.cs:70 in fa2f142. [](commit_id = fa2f142624d2c2c15688c7ab678c7ae591dcd73f, deletion_comment = False)

jennyf19 commented:
sure. i'll sync w/petro tomorrow on the cache specific comments to see if we can make them more meaningful.

---
In reply to: [409092788](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1154#issuecomment-409092788) [](ancestors = 409092788)

jennyf19 committed:
more precise log messages part1

jennyf19 committed:
Merge branch 'jennyf/coreLogging' of https://github.com/AzureAD/azure-activedirectory-library-for-dotnet into jennyf/coreLogging

bgavrilMS commented on msal/src/Microsoft.Identity.Client/Internal/Requests/RequestBase.cs:
```diff
@@ -169,13 +169,15 @@ protected void ValidateScopeInput(SortedSet<string> scopesToValidate)
             {
                 apiEvent.ApiErrorCode = ex.ErrorCode;
-                AuthenticationRequestParameters.RequestContext.Logger.Error(ex);
+                string noPiiMsg = MsalExceptionFactory.GetPiiScrubbedExceptionDetails(ex);
```
I believe there is an overload of Error that takes in an exception. It could be modified to get the scrubbed details as well.

jennyf19 replied:
@bgavrilMS That's a good idea. I think we might do some refactoring of the logging, so that's something to consider. thanks.

jennyf19 committed:
Merge remote-tracking branch 'origin/dev' into jennyf/coreLogging

jennyf19 committed:
Address binary breaking changes:
- adding setter for `ServiceCodeErrors`
- fix enum values for `PromptBehavior`

bgavrilMS reviewed (approved):
LGTM

jennyf19 commented:
@henrik-me I believe I addressed all the comments. I checked the dll's and fixed a few issues w/adal. Have one more, but will do that in a separate PR.

jennyf19 committed:
address spelling

bgavrilMS reviewed (approved):
LGTM
